DuckDuckGo/RemoteMessaging.swift

```diff
@@ -173,6 +177,20 @@ struct RemoteMessaging {
                 if case let .success(subscription) = subscriptionResult {
                     privacyProDaysSinceSubscribed = Calendar.current.numberOfDaysBetween(subscription.startedAt, and: Date()) ?? -1
                     privacyProDaysUntilExpiry = Calendar.current.numberOfDaysBetween(Date(), and: subscription.expiresOrRenewsAt) ?? -1
+                    privacyProPurchasePlatform = subscription.platform.rawValue
+
+                    switch subscription.status {
+                    case .autoRenewable, .gracePeriod:
+                        privacyProIsActive = true
+                    case .notAutoRenewable:
+                        privacyProIsActive = true
+                        privacyProIsExpiring = true
```

Comment:
It's deliberate that both of these are set here - `notAutoRenewable` is the state when a user has cancelled their subscription but it's still active while they wait out the remainder of the sub. We still want to consider them active here since they can still use Privacy Pro.